sktime/forecasting/compose/_skforecast_reduce.py

```diff
@@ -0,0 +1,380 @@
+    References
+    ----------
+    .. [1]
+        https://skforecast.org/latest/api/forecasterautoreg#forecasterautoreg
+
+    Examples
+    --------
+    >>> from sktime.forecasting.compose import SkforecastAutoreg
+
+    Without exogenous features
+
+    >>> from sklearn.linear_model import LinearRegression
+    >>> from sktime.datasets import load_airline
+    >>> y = load_airline()
+    >>> forecaster = SkforecastAutoreg(  # doctest: +SKIP
+    ...     LinearRegression(), 2
+    ... )
+    >>> forecaster.fit(y)  # doctest: +SKIP
+    SkforecastAutoreg(lags=2, regressor=LinearRegression())
+    >>> y_pred = forecaster.predict(fh=[1, 2, 3])  # doctest: +SKIP
+    >>> y_pred_int = forecaster.predict_interval(  # doctest: +SKIP
+    ...     fh=[2], coverage=[0.9, 0.95]
+    ... )
+    >>> y_pred_qtl = forecaster.predict_quantiles(  # doctest: +SKIP
+    ...     fh=[1, 3], alpha=[0.8, 0.3, 0.2, 0.7]
+    ... )
+
+    With exogenous features
+
+    >>> from sklearn.ensemble import RandomForestRegressor
+    >>> from sktime.datasets import load_longley
+    >>> y, X = load_longley()
+    >>> y = y.reset_index(drop=True)  # period index is not supported in ``skforecast``
+    >>> X = X.reset_index(drop=True)  # period index is not supported in ``skforecast``
+    >>> y_train = y.head(n=12)
+    >>> y_test = y.tail(n=4)
+    >>> X_train = X.head(n=12)
+    >>> X_test = X.tail(n=4)
+    >>> forecaster = SkforecastAutoreg(  # doctest: +SKIP
+    ...     RandomForestRegressor(), [2, 4]
+    ... )
+    >>> forecaster.fit(y_train, X=X_train)  # doctest: +SKIP
+    SkforecastAutoreg(lags=[2, 4], regressor=RandomForestRegressor())
+    >>> y_pred = forecaster.predict(fh=[1, 2, 3], X=X_test)  # doctest: +SKIP
+    >>> y_pred_int = forecaster.predict_interval(  # doctest: +SKIP
+    ...     fh=[1, 3], X=X_test, coverage=[0.6, 0.4]
+    ... )
+    >>> y_pred_qtl = forecaster.predict_quantiles(  # doctest: +SKIP
+    ...     fh=[1, 3], X=X_test, alpha=[0.01, 0.5]
+    ... )
+
+    """
+
+    _tags = {
+        "authors": ["yarnabrina"],
+        "maintainers": ["yarnabrina"],
+        "y_inner_mtype": "pd.Series",
+        "X_inner_mtype": "pd.DataFrame",
+        "requires-fh-in-fit": False,
+        "enforce_index_type": [pandas.DatetimeIndex, pandas.RangeIndex],
+        "handles-missing-data": False,
+        "capability:insample": False,
+        "capability:pred_int": True,
+        "capability:pred_int:insample": False,
+        "python_version": ">=3.8,<3.13",
+        "python_dependencies": ["skforecast>=0.11"],
+    }
+
+    def __init__(
+        self: "SkforecastAutoreg",
+        regressor: object,
+        lags: typing.Union[int, numpy.ndarray, list],
+        transformer_y: typing.Optional[object] = None,
+        transformer_exog: typing.Optional[object] = None,
+        weight_func: typing.Optional[typing.Callable] = None,
+        differentiation: typing.Optional[int] = None,
+        fit_kwargs: typing.Optional[dict] = None,
+    ) -> None:
```

Comment:
As per the [docs](https://skforecast.org/0.12.1/api/forecasterautoreg), they support `binner_kwargs` too.

Reply:
Will check it out.

Reply:
I've added all missing args now.